scc/cert/serialization/types_test.go

```diff
@@ -0,0 +1,101 @@
+package serialization
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHexBytes_MarshalJSON(t *testing.T) {
+	h := HexBytes([]byte{0x01, 0x2a, 0xbc})
+	data, err := h.MarshalJSON()
+	require.NoError(t, err)
+	require.Equal(t, []byte(`"0x012abc"`), data)
+	h = nil
+	data, err = h.MarshalJSON()
+	require.NoError(t, err)
+	require.Equal(t, []byte("null"), data)
+}
+
+func TestHexBytes_UnmarshalJSON_ValidHexString(t *testing.T) {
+	var h HexBytes
+	data := []byte(`"0x012abc"`)
+	require.NoError(t, h.UnmarshalJSON(data))
+	data = []byte(`"null"`)
+	require.NoError(t, h.UnmarshalJSON(data))
+	data = []byte(`"0x12abc"`)
+	require.NoError(t, h.UnmarshalJSON(data))
+	require.Equal(t, HexBytes([]byte{0x1, 0x2a, 0xbc}), h)
+}
+
+func TestHexBytes_UnmarshallJSON_Invalid(t *testing.T) {
+	var h HexBytes
+	data := []byte(`"0xg"`)
+	err := h.UnmarshalJSON(data)
+	require.Error(t, err)
+	data = []byte(`"01"`)
+	err = h.UnmarshalJSON(data)
+	require.Error(t, err)
+}
+
+func TestHexBytes_String(t *testing.T) {
+	h := HexBytes([]byte{0x01, 0x2a, 0xbc})
+	require.Equal(t, "0x012abc", h.String())
+}
+
+func TestHexBytes48_MarshalJSON(t *testing.T) {
+	p := HexBytes48([48]byte{0x01})
+	data, err := p.MarshalJSON()
+	require.NoError(t, err)
+	expected := `"0x010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"`
+	require.Equal(t, []byte(expected), data)
+}
+
+func TestHexBytes48_UnmarshalJSON_ShortHexString(t *testing.T) {
```

Comment:
By our test name [convention](https://github.com/Fantom-foundation/Tosca/issues/883) the name of a test like this should also state the intended behavior. Something like
```suggestion
func TestHexBytes48_UnmarshalJSON_TooShortHexStringIsRejected(t *testing.T) {
```

The other names of tests in this file would benefit from a similar renaming.